app/lib/activitypub/activity/create.rb

```diff
@@ -237,8 +237,11 @@ def process_attachments
       next if attachment['url'].blank? || media_attachments.size >= 4
 
       begin
-        href             = Addressable::URI.parse(attachment['url']).normalize.to_s
-        media_attachment = MediaAttachment.create(account: @account, remote_url: href, description: attachment['summary'].presence || attachment['name'].presence, focus: attachment['focalPoint'], blurhash: supported_blurhash?(attachment['blurhash']) ? attachment['blurhash'] : nil)
+        href           = Addressable::URI.parse(attachment['url']).normalize.to_s
+        thumbnail_href = attachment['icon'].is_a?(String) ? attachment['icon'] : attachment.dig('icon', 'url')
+        thumbnail_href = Addressable::URI.parse(thumbnail_href).normalize.to_s if thumbnail_href.present?
```

Comment:
It's probably not a big deal, but I'd prefer if the thumbnail not having a valid URL would not make the attachment itself fail to process.

Reply:
Welp, I ended up refactoring `Remotable` to make this happen. Now both attachments are set to *not* download-on-assign, so when you assign the URLs, no side effects happen. Then I made `download_file!`/`download_thumbnail!` methods do the actual heavy lifting (aliased to `reset_file!` etc, because it's a better name for what it was doing), and those are methods are what gets called automatically when download-on-assign is desired.